cpp/tests/copying/concatenate_tests.cpp

```diff
@@ -226,6 +197,53 @@ TEST_F(StringColumnTest, ConcatenateTooLarge)
   EXPECT_THROW(cudf::concatenate(input_cols), std::overflow_error);
 }
 
+TEST_F(StringColumnTest, ConcatenateLargeStrings)
+{
+  CUDF_TEST_ENABLE_LARGE_STRINGS();
+  auto itr = thrust::constant_iterator<std::string_view>(
+    "abcdefghijklmnopqrstuvwxyABCDEFGHIJKLMNOPQRSTUVWXY");                // 50 bytes
+  auto input = cudf::test::strings_column_wrapper(itr, itr + 5'000'000);  // 250MB
+  std::vector<cudf::column_view> input_cols;
+  for (int i = 0; i < 10; ++i) {  // 2500MB > 2GB
+    input_cols.push_back(input);
+  }
+  auto result = cudf::concatenate(input_cols);
+  auto sv     = cudf::strings_column_view(result->view());
+  EXPECT_EQ(sv.size(), 50'000'000);
+  EXPECT_EQ(sv.offsets().type(), cudf::data_type{cudf::type_id::INT64});
+
+  // verify results in sections
+  auto splits = std::vector<cudf::size_type>({5'000'000,
```

Comment:
Could we generate the values of the splits in the same for loop as `input_cols` above? Just trying to reduce hard-coded values a bit.